pkg/util/azureclient/mgmt/authorization/denyassignment_addons.go

```diff
@@ -30,3 +38,50 @@ func (c *denyAssignmentClient) ListForResourceGroup(ctx context.Context, resourc
 
 	return result, nil
 }
+
+func (c *denyAssignmentClient) DeleteDenyAssignment(ctx context.Context, fpTokenCredential *azidentity.ClientCertificateCredential, subscriptionDoc *api.SubscriptionDocument, doc *api.OpenShiftClusterDocument) error {
+	deleteClient, err := NewDenyAssignmentsARMClient(subscriptionDoc.ID, fpTokenCredential, nil)
+	if err != nil {
+		return err
+	}
+
+	managedResourceGroupName := strings.Split(doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, "/")[4]
+	denyAssignmentList, err := c.ListForResourceGroup(ctx, managedResourceGroupName, "")
+	if err != nil {
+		return err
+	}
+
+	// There should never be more than one deny assignment, but in case there is, don't proceed
+	if len(denyAssignmentList) > 1 {
+		return nil
```

Comment:
I'm not sure how we would ever end up with more than one deny assignment, but just in case, I think it would be wise to return an error here to ensure something gets logged.